specs/rigil/precompiles.md

```diff
@@ -220,6 +222,31 @@ Signs an Ethereum Transaction, 1559 or Legacy, and returns raw signed transactio
 function signEthTransaction(bytes memory txn, string memory chainId, string memory signingKey) view returns (bytes memory)
 ```
 
+### `ensureTxnValid`
+
+
+[🔗 Implementation - WIP]()
+
+Address: `0x0000000000000000000000000000000040100003`
+
+Reverts if the input `bytes` do not contain a well-formed, RLP-encoded transaction.
```

Comment:
in our discussion this also checks if nonces are correct as well? well formed only implies types are correct I believe? Also I think we should kill 2 birds with 1 stone and return simulation results with this precompile as well. or happy to merge and add in the future

Reply:
why make this a precompile rather than use a solidity library? 
https://github.com/bakaoh/solidity-rlp-encode encode 
https://github.com/hamdiallam/Solidity-RLP decode

Reply:
@amiller I'm actually leaning towards exposing `abidecode` and `abiencode` as precompiles.  From a DevEx perspective, I feel like I'm expending a fair bit of energy worrying about (de)serializing RLP.  It seems like we should offer a fast, stable API for this, similar to `toJSON`/`fromJSON` in the browser.

Do we lose anything by moving in this direction?

Reply:
abi.encode and abi.decode are already built into Solidity. RLPEncode/Decode have If the rationale for making something a precompile is performance then it makes sense to quantify the before/after relative to existing libraries performance.

ToJSON and FromJSON i think there are not even Solidity libraries for it so it seems most justified

Reply:
Ah ok, agreed.  That makes much more sense.

I'll just remove `ensureTxValid` entirely.